src/gui_common/tooltip/ToolTipCallbackData.cs

```diff
@@ -23,19 +28,43 @@ public void OnMouseEnter()
 
     public void OnMouseExit()
     {
-        ToolTipManager.Instance.MainToolTip = null;
-        ToolTipManager.Instance.Display = false;
+        // This used to always unset the main tooltip, but this now only unsets the tooltip if no one else had touched
+        // it in the meantime
+        if (ToolTipManager.Instance.MainToolTip == ToolTip)
+        {
+            ToolTipManager.Instance.MainToolTip = null;
+            ToolTipManager.Instance.Display = false;
+        }
     }
 
     public void OnExitingTree()
     {
         OnMouseExit();
 
-        // This is to avoid premature unregistration
-        if (!ToolTipable.IsReParenting())
+        // To allow nodes with tooltips to exit and re-enter the scene, tooltips are now only unregistered by the code
+        // that registered them. Or if using auto unregistering, then this is also done
+        if (AutoUnregisterOnTreeExit)
+            Unregister();
```

Comment:
Well not really, because for reparenting reasons a thing shouldn't remove the tooltip registrations unless it is not going to be used anymore. So for example a custom control that releases its tooltips when it is deleted, would incorrectly lead to warnings printed here if that node was reparented. That's why I added the PostShutdownActions to detect the condition of forgotten tooltips.

Reply:
> That's why I added the PostShutdownActions to detect the condition of forgotten tooltips.

I guess that's good enough.